### ISSUE TITLE
Scale image width for tie-point extraction

### ISSUE THREAD

**Issue by kikislater** [OpenDroneMap/ODM#983]
### How did you install OpenDroneMap? (Docker, natively, ...)?

Docker

### What's your browser and operating system? (Copy/paste the output of https://www.whatismybrowser.com/)

Firefox 67 on Linux

### What is the problem?

Scale image width for tie-point extraction does not respect standard in other photogrammetry software

### What should be the expected behavior? If this is a feature request, please describe in detail the changes you think should be made to the code, citing files and lines where changes should be made, if possible.

Default image size is 800px ! So why it's 800px when in others popular software like Photoscan or Pix4d it's like that : 
High : original image / 2
Medium : original image / 4
Low : original image / 8 

Note : May be it was discussed before but didn't find topic about it ! 

### How can we reproduce this? (What steps did you do to trigger the problem? What parameters are you using for processing? If possible please include a copy of your dataset uploaded on Google Drive or Dropbox. Be detailed)

Default settings

**Comment by pierotofy** [comment]
Hey @kikislater, can you expand on the default image size? Are you referring to the `--resize-to` parameter? (By default it's actually 2048 so you have me confused there).

**Comment by kikislater** [comment]
Here : https://github.com/dronemapper-io/NodeMICMAC/pull/6/files#diff-843711f7393fe9a6e9f11ba1c8132afb

> "help": "Scale image width for tie-point extraction. Default: **800** 

odm_options.json, I thought it was inherit from ODM ! Even 2048, why this choice and not image size / 2 or whatever ... ?!

**Comment by smathermather** [comment]
Well, you raise a good point. I think 2048 is the default resolution based on _nuisance preventative development_, i.e. most users get a pretty good result on the less than adequate hardware they bring to bear for running ODM. We get lots fewer bug reports from folks running out of memory at 2048.

So, do we make an API change to comply with the general standard? If so, this becomes a feature request.

**Comment by smathermather** [comment]
And then we may have a separate standard for NodeMICMAC based on best practices with that... .

**Comment by kikislater** [comment]
I'm not that kind of user but if you think user : they don't care about pixels to set in photogrammetry software. They watch high, medium and low and most are pretty happy with that ... That's the minding of my post

**Comment by pierotofy** [comment]
Ah yes, makes sense. Wonder if this should have been opened under the NodeMICMAC repo then.

**Comment by smathermather** [comment]
Well, it applies to both. :)

**Comment by pierotofy** [comment]
I see.

One problem is that ODM accepts images, from multiple cameras (which might be of different resolution) but OpenSfM wants to know a value in pixels. We could take a maximum I suppose.

**Comment by kikislater** [comment]
Hum ok for OpenSfM. 
Watch what they do in Pix4d (image frome internet) => 
![image](https://user-images.githubusercontent.com/13238064/58377993-4f165480-7f9c-11e9-97ef-d2571d3a968f.png)
In Photoscan (high, medium, low told before) : 
![image](https://user-images.githubusercontent.com/13238064/58378021-b6340900-7f9c-11e9-8bab-4e3e41ea6b81.png)


And it's a purpose also for another thing : most photogrammetric technics differ from Dem/Ortho computation vs "pure" 3D model. Users like not to worry about parameters and others software purpose some interactive interface to process without knowing anything. I like 2 first images from ODM feature list. Just a purpose to process Ortho/DEM only or 3 models like others do (Photoscan is the same but less interactive, intuitive interface with icon is nice I think : could be discuss. I guess my thread is about discussing not "a do that !" or missing thing) : 

![image](https://user-images.githubusercontent.com/13238064/58377988-2db56880-7f9c-11e9-9b1e-d0d97a8e8bc6.png)

![image](https://user-images.githubusercontent.com/13238064/58378033-057a3980-7f9d-11e9-9438-47af5170ce33.png)

![image](https://user-images.githubusercontent.com/13238064/58378155-e630db80-7f9f-11e9-8b60-467897461f1a.png)
![image](https://user-images.githubusercontent.com/13238064/58378200-fdbc9400-7fa0-11e9-9ff0-ef3f23ae5453.png)

**Comment by smathermather** [comment]
OpenSfM may want a specific resolution, but the idea of _high_, _medium_, _low_ choices could be a wrapper for that, and perform similarly to other software as above:
```
High : original image / 2
Medium : original image / 4
Low : original image / 8
```

I never new what these meant for Agisoft or Pix4D, but it makes sense.

**Comment by kikislater** [comment]
If you want reference for Agisoft high, medium, low ... 
https://www.agisoft.com/forum/index.php?topic=1741.0

> Images are downscaled corresponding to the target quality selected:
> Ultra High - no scaling
> High - images are downscaled two times by each side
> Medium - downscaling four times by each side
> Low - eight times
> Lowest - sixteen times

In others words : 

> Ultra = full size
> High = Downscaled to 50%
> Medium = Downscaled to 25%
> Low = Downscaled to 12,5%
> Lowest = Downscaled to 6,75%

**Comment by pierotofy** [comment]
This will be included as part of #1156

**Comment by pierotofy** [comment]
This has been implemented! :clinking_glasses:
